CMakeLists.txt

```diff
@@ -132,9 +132,22 @@ endif()
 option(PSMOVE_USE_LOCAL_OPENCV "Use locally-built OpenCV (static linking)" OFF)
 if (PSMOVE_USE_LOCAL_OPENCV)
     # Force tracker build with a locally-built opencv library installation
-    set(OPENCV_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/opencv)
-    link_directories(${OPENCV_SRC_DIR}/build/lib)
-    link_directories(${OPENCV_SRC_DIR}/build/3rdparty/lib)
+    
+    # allow for a custom OPENCV_SRC_DIR to be passed
+    if (NOT DEFINED OPENCV_SRC_DIR)
+        set(OPENCV_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/opencv)
+    endif()
+
+    # allow for an in-source build of OpenCV
+    if (DEFINED OPENCV_BUILT_IN_SOURCE AND OPENCV_BUILT_IN_SOURCE)
```

Comment:
Not sure if OpenCV in-source builds work (but I guess you have tested this?), in any case this can be a boolean option:

```
option(PSMOVE_BUILD_OPENCV_IN_SOURCE "Build the local OpenCV in its source tree" OFF)
```

And then:

```
if (PSMOVE_BUILD_OPENCV_IN_SOURCE)
   ...
else()
   ...
endif()
```

This way, it's easier for other developers to discover these options, they don't have to read the `CMakeLists.txt` source then.